pyro/infer/trace_mean_field_elbo.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import absolute_import, division, print_function
+
+import warnings
+
+from torch.distributions import kl_divergence
+
+from pyro.distributions.util import is_identically_zero, scale_and_mask
+from pyro.infer.trace_elbo import Trace_ELBO
+from pyro.infer.util import is_validation_enabled, torch_item
+from pyro.util import warn_if_nan
+
+
+def _check_mean_field_requirement(model_trace, guide_trace):
+    """
+    Checks that the guide and model sample sites are ordered identically.
+    This is sufficient but not necessary for correctness.
+    """
+    model_sites = [name for name, site in model_trace.nodes.items()
+                   if site["type"] == "sample" and name in guide_trace.nodes]
+    guide_sites = [name for name, site in guide_trace.nodes.items()
+                   if site["type"] == "sample" and name in model_trace.nodes]
+    assert set(model_sites) == set(guide_sites), "programmer error"
```

Comment:
'programmer error' seems a potentially confusing assert msg, no?

Reply:
if find it standard, and it should never be surfaced to users, but ok I'll remove it.